src/django/api/reports.py

```diff
@@ -43,21 +46,28 @@ def monthly_promoted_name_and_address():
 
 def geocoding_time_without_queue(date_format):
     temp_data = dict()
-    listitems = FacilityListItem.objects.order_by(
-        '-created_at').values_list('processing_results', flat=True)[:1000]
-    for processing_results in listitems:
-        for r in processing_results:
-            if r['action'] == ProcessingAction.GEOCODE:
-                try:
-                    started = try_parsing_date(r['started_at'])
-                    finished = try_parsing_date(r['finished_at'])
-                    date = started.strftime(date_format)
-                    time = finished - started
-                    counts = temp_data.get(date, (0, 0))
-                    temp_data[date] = (counts[0] + 1,
-                                       counts[1] + time.total_seconds())
-                except ValueError:
-                    pass
+
+    start_date = datetime.now(tz=timezone.utc) - relativedelta(months=4)
+    processing_results = FacilityListItem.objects.filter(
+        created_at__gte=start_date,
+        processing_results__contains=[{"action": "geocode"}]
```

Comment:
This line filters out FacilityListItems that have *no* processing results with `action: 'geocode'`.